strawberryfields/configuration.py

```diff
@@ -15,138 +15,229 @@
-
-    def __init__(self, name="config.toml"):
-        # Look for an existing configuration file
-        self._config = DEFAULT_CONFIG
-        self._config_file = {}
-        self._filepath = None
-        self._name = name
-        self._user_config_dir = user_config_dir("strawberryfields", "Xanadu")
-        self._env_config_dir = os.environ.get("SF_CONF", "")
-
-        # Search the current directory, the directory under environment
-        # variable SF_CONF, and default user config directory, in that order.
-        directories = [os.getcwd(), self._env_config_dir, self._user_config_dir]
-        for directory in directories:
-            self._filepath = os.path.join(directory, self._name)
-            try:
-                config = self.load(self._filepath)
-                break
-            except FileNotFoundError:
-                config = False
-
-        if config:
-            self.update_config()
-        else:
-            log.info("No Strawberry Fields configuration file found.")
-
-    def update_config(self):
-        """Updates the configuration from either a loaded configuration
-        file, or from an environment variable.
-
-        The environment variable takes precedence."""
-        for section, section_config in self._config.items():
-            env_prefix = "SF_{}_".format(section.upper())
-
-            for key in section_config:
-                # Environment variables take precedence
-                env = env_prefix + key.upper()
-
-                if env in os.environ:
-                    # Update from environment variable
-                    self._config[section][key] = parse_environment_variable(env, os.environ[env])
-                elif self._config_file and key in self._config_file[section]:
-                    # Update from configuration file
-                    self._config[section][key] = self._config_file[section][key]
-
-    def __getattr__(self, section):
-        if section in self._config:
-            return self._config[section]
-
-        raise ConfigurationError("Unknown Strawberry Fields configuration section.")
-
-    @property
-    def path(self):
-        """Return the path of the loaded configuration file.
-
-        Returns:
-            str: If no configuration is loaded, this returns ``None``."""
-        return self._filepath
-
-    def load(self, filepath):
-        """Load a configuration file.
-
-        Args:
-            filepath (str): path to the configuration file
-        """
-        with open(filepath, "r") as f:
-            self._config_file = toml.load(f)
-
-        return self._config_file
-
-    def save(self, filepath):
-        """Save a configuration file.
-
-        Args:
-            filepath (str): path to the configuration file
-        """
-        with open(filepath, "w") as f:
-            toml.dump(self._config, f)
+DEFAULT_CONFIG = create_config_object()
+configuration = load_config()
+config_file_path = look_for_config_in_file()[1]
```

Comment:
Perhaps this indicates that `look_for_config_in_file()` should simply be changed to `get_config_filepath()`, and only return the filepath? Them `load_config()` can be called manually.

Reply:
That is really amazing comment, thank you so much! I've updated the function names, logic and tests accordingly :)